R/step-subset.R

```diff
@@ -117,7 +119,11 @@ dt_call.dtplyr_step_subset <- function(x, needs_copy = x$needs_copy) {
 
   if (length(x$on) > 0) {
     out$on <- call2(".", !!!syms(x$on))
-    out$allow.cartesian <- TRUE
+    if (is.null(x$allow_cartesian)) {
+      out$allow.cartesian <- TRUE
+    } else {
+      out$allow.cartesian <- parent$allow_cartesian
+    }
```

Comment:
```suggestion
    out$allow.cartesian <- parent$allow_cartesian %||% TRUE
```
?

Reply:
I ended up adding an explicit `allow_cartesian = TRUE` to `step_subset_on()` and it made this part much more straightforward